app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,85 @@
+require 'miq_apache'
+
+class MiqCockpitWsWorker < MiqWorker
+  require_nested :Runner
+
+  APACHE_CONF_FILE = '/etc/httpd/conf.d/manageiq-cockpit.conf'.freeze
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_start_cockpit_ws
+    @supports_cockpit_ws ||= MiqEnvironment::Command.is_linux? &&
+                             File.exist?('/usr/libexec/cockpit-ws') &&
+                             File.exist?('/usr/libexec/cockpit-ssh')
+  end
+
+  def self.can_activate_cockpit?(opts = nil)
+    return false unless has_required_role?
+
+    opts ||= MiqCockpitWsWorker.worker_settings
+    opts[:external_url] ? true : can_start_cockpit_ws
+  end
+
+  def self.sync_workers
+    opts = MiqCockpitWsWorker.worker_settings
+    install_apache_proxy_config(opts) if MiqEnvironment::Command.supports_apache?
+    can_activate = can_activate_cockpit?(opts)
+    @workers = opts[:external_url] || can_activate ? 1 : 0
+    super
+  end
+
+  def self.install_apache_proxy_config(opts = nil)
+    opts ||= MiqCockpitWsWorker.worker_settings
+    if has_required_role? || opts[:external_url]
+      url = MiqCockpit::WS.url(APACHE_CONF_FILE)
+    elsif MiqServer.my_server.has_active_role?(MiqUiWorker.required_roles)
+      url = MiqRegion.my_region.remote_cockpit_ws_url
+    end
+
+    if url
+      MiqCockpit::ApacheConfig.new(url).save(APACHE_CONF_FILE)
+    elsif File.exist?(APACHE_CONF_FILE)
+      File.truncate(APACHE_CONF_FILE, 0)
+    end
+    MiqServer.my_server.queue_restart_apache
+  end
+
+  # Copied from old MiqProcess method
+  def get_child_pids(pid = nil)
+    pid ||= Process.pid
+
+    result = []
+
+    # Gets processes whose parent pid is pid
+    begin
+      output = MiqUtil.runcmd("ps -l --ppid #{pid}")
```

Comment:
Sys::ProcTable has `ppid` method on each of the structs returned from the `ps` command, please try using that.

https://github.com/djberg96/sys-proctable/blob/2bc2fb9b42e8c1c8effd53ced186e1afb7c33428/lib/linux/sys/proctable.rb#L192